pixelluxe-gui/src/main/java/at/jku/pixelluxe/ui/App.java

```diff
@@ -194,6 +194,16 @@ private void onConvolutionPressed() {
 		});
 	}
 
+	public void update(PaintableImage paintableImage) {
+		this.model.updateAndGet(newmodel -> {
+			int selectedImage = this.selectedImage;
+			if (selectedImage < 0 || selectedImage >= newmodel.imageFiles().size()) {
+				return newmodel;
+			}
+			return newmodel.with(selectedImage, new ImageFile(paintableImage, newmodel.imageFiles().get(selectedImage).backingFile()));
+		});
+	}
+
 	private void updatePaintable(PaintableImage paintableImage) {
```

Comment:
Do drin könntest de neue update verwenden. Sonst Code-Duplication :). 

Eigentlich müsste ma de beiden Methoden aber renamen. Aktuell is is ziemlich unklar, wo de Unterschiede liegen. Zb `updateSelected` und `updateAndRepaintSelected` oda so.